internal/plan/plan.go

```diff
@@ -184,6 +187,76 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// GetCommand overrides the default arguments (inside [ ]) if
+// CmdArgs is present and returns the command as a stream of strings.
+func (s *Service) GetCommand() ([]string, error) {
+	args, err := shlex.Split(s.Command)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse service %q command: %s", s.Name, err)
+	}
+	fargs := make([]string, 0)
+	for _, arg := range args {
+		if arg == "[" || arg == "]" {
+			if len(s.CmdArgs) > 0 {
+				break
+			} else {
+				continue
```

Comment:
I'm pretty sure this isn't the logic we need: if there are `CmdArgs` we should insert them at this point but then keep appending the remaining args, right? For example:

```
command: echo foo [ bar ] baz
```

With `--args arg1 arg2 ;` should yield `echo foo arg1 arg2 baz`, right?

We should also handle various error cases like multiple `[ ... ]` groups, attempts to nest, etc (should probably be disallowed).

I had a quick play with it and came up with https://go.dev/play/p/CH25ToKC1q1 (but improvements/rewrite welcome).

Reply:
Oh, I see you're explicitly only allowed `[ ... ]` at the end of the command. Sorry I'd missed that -- is that by design?

Reply:
Thanks for the implementation. Yes, allowing `[ ... ]` at the very end was by design. 
I was initially considering support for `echo foo [ bar ] baz` as well, but had a quick chat within the team and `[ ... ]` at the end seemed to be the spec.
Let me know if you think `echo foo [ bar ] baz` would be a better idea and we can chat about it.
Cool implementation as well, thanks! I am adding some replies in the `checkCommand` comments.

Reply:
I do think the `[ ... ]` syntax makes me think it'd work anywhere in the command, because there's a start and an end character. If we have an arbitrary restriction ("it only works at the end") then we have document that as a restriction. Unless there's a good reason to restrict it to only the end, I don't think we should restrict it (I thought it was a bug).

Reply:
Indeed, optional args were considered only at the end, as a sort of analogy to ENTRYPOINT + CMD.

I agree that it could work anywhere, and while this initial design choice falls on the conservative side, I do not see any major problems about allowing it to be anywhere in the `command`. I'm a bit sceptical about allowing `[...]` to be a prefix though, but apart from that, y, I guess it could be anywhere in the code

Reply:
Okay, that's fine -- if we're going that route let's just make it clear in the help. We can always make it less conservative later if a need arises (but harder to go the other way!).

Oh, that reminds me, we should also update the `command` field in README.md under [Layer specification](https://github.com/canonical/pebble/#layer-specification) to mention the new `[ ... args ... ]` feature.

Reply:
This should be resolved now, but would be glad if you took another look at it. :D